src/app/client-view/toggle.weight.service.ts

```diff
@@ -0,0 +1,13 @@
+export class ToggleWeightService {
+  weightNo: number;
```

Comment:
```suggestion
  weightNo: number;
  displayedWeightNo: number;
```
Another variable is needed to avoid overwriting the weight in the methods below